src/Domain/DonationRepository.php

```diff
@@ -459,6 +459,35 @@ public function findReadyToClaimGiftAid(bool $withResends): array
         return $qb->getQuery()->getResult();
     }
 
+    /**
+     * @return Donation[]
+     */
+    public function findNotFullyMatchedToCampaignsWhichClosedSince(DateTime $closedSinceDate): array
+    {
+        $now = (new DateTime('now'));
+        $qb = $this->getEntityManager()->createQueryBuilder()
```

Comment:
probably not worth changing now, but in general I think it's better to use DQL rather than query builder. As [Doctrine docs say](https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/query-builder.html):
> The QueryBuilder is not an abstraction of DQL, but merely a tool to dynamically build it. You should still use plain DQL when you can, as it is simpler and more readable. More about this in the [FAQ](https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/faq.html).

Reply:
Interesting @bdsl, will consider for next time. If we do want to mock pieces of these methods in tests it would also be less cumbersome with one call.